pages/items/[id]/index.js

```diff
@@ -0,0 +1,45 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+import ItemDetails from "@/components/ItemDetails";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const { isReady } = router;
+  const { id } = router.query;
+
+  const { data: item, mutate, isLoading, error } = useSWR(`/api/items/${id}`);
+
+  async function handleStatus() {
+    const newItem = { ...item, inDiscuss: !item.inDiscuss };
+    await fetch(`/api/items/${id}`, {
+      method: "PUT",
+      body: JSON.stringify(newItem),
+    });
+    mutate(newItem);
```

Comment:
You should use the empty mtutate call here `mutate()`. That results in a refetch by the useSWR hook which gets you the new updated item. By using newItem here you ignore possible changes in your backend to the item content. Even though right now you know what is happening, this might change in the future. So refetching is the more robust solution.

Reply:
Thanks!